app/commands/generate_missing_mappings_command.rb

```diff
@@ -0,0 +1,318 @@
+        in terms of product type, function, or purpose.
+        For example:
+          - "Apparel & Accessories" and "Clothing, Shoes & Jewelry"
+          - "Apparel & Accessories > Clothing > One-Pieces" and "Clothing, Shoes & Accessories > Women > Women's Clothing > Jumpsuits & Rompers"
+      2. Mark a mapping as No, i.e. incorrect, if two categories of a mapping are irrevant to each other
+        in terms of product type, function, or purpose.
+        For example:
+          - "Apparel & Accessories > Clothing > Dresses" and "Clothing, Shoes & Jewelry>Shoe, Jewelry & Watch Accessories"
+          - "Apparel & Accessories" and "Clothing, Shoes & Jewelry>Luggage & Travel Gear"
+      Note, the character ">" in a category name indicates the start of a new category level. For example:
+      "sporting goods > exercise & fitness > cardio equipment"'s ancestor categories are "sporting goods > exercise & fitness" and "sporting goods".
+      You will receive a list of mappings. Each mapping contains a from_category name and a to_category name.
+      e.g. user's prompt in json format:
+      [
+        {
+          "from_category_id": "111",
+          "from_category": "Apparel & Accessories > Jewelry > Smart Watches",
+          "to_category_id": "222",
+          "to_category": "Clothing, Shoes & Jewelry>Men's Fashion>Men's Watches>Men's Smartwatches",
+          },
+        {
+          "from_category_id": "333",
+          "from_category": "Apparel & Accessories > Clothing > One-Pieces",
+          "to_category_id": "444",
+          "to_category": "Clothing, Shoes & Accessories > Women > Women's Clothing > Outfits & Sets",
+          },
+      ]
+      You evaluate accuracy of every mapping and reply in the following format. Do not change the order of mappings in your reply.
+      e.g. your response in json format:
+      [
+        {
+          "from_category_id": "111",
+          "from_category": "Apparel & Accessories > Jewelry > Smart Watches",
+          "to_category_id": "222",
+          "to_category": "Clothing, Shoes & Jewelry>Men's Fashion>Men's Watches>Men's Smartwatches",
+          "agree_with_mapping": "Yes",
+        },
+        {
+        "from_category_id": "333",
+          "from_category": "Apparel & Accessories > Clothing > One-Pieces",
+          "to_category_id": "444",
+          "to_category": "Clothing, Shoes & Accessories > Women > Women's Clothing > Outfits & Sets",
+          "agree_with_mapping": "No",
+        },
+      ]
+    CONTEXT
+  end
+
+  def initialize_clients
+    frame("Initializing clients for performing semantic search") do
+      @openai_client = create_openai_client
+      @qdrant_client = create_qdrant_client
+    end
+  end
+
+  def create_openai_client
+    OpenAI::Client.new(
+      access_token: ENV["OPENAI_API_KEY"],
+      uri_base: "https://openai-proxy.shopify.ai/v1",
+      request_timeout: 10,
+    )
+  end
```

Comment:
I wonder if cached methods would flow nicer vs a setup call? eg

```suggestion
  def openai_client
    @openai_client ||= OpenAI::Client.new(
      access_token: ENV["OPENAI_API_KEY"],
      uri_base: "https://openai-proxy.shopify.ai/v1",
      request_timeout: 10,
    )
  end
```

Reply:
Could even put a spinner in here if we wanted one
```suggestion
  def openai_client
    return @openai_client if @openai_client
    
    spinner.spin(...) do
      @openai_client = OpenAI::Client.new(
        access_token: ENV["OPENAI_API_KEY"],
        uri_base: "https://openai-proxy.shopify.ai/v1",
        request_timeout: 10,
      )
    end
  end
```

Reply:
switched to use cached methods